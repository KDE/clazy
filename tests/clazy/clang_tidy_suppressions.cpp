#include <QtCore/QString>
#include <QtCore/QList>
#include <QtCore/QDateTime>

void testClangTidySuppressionsSameLine()
{
    QString s = "linting for line disabled"; // NOLINT 
    if (s == "supressed nonexistent") {} // NOLINT(clazy-bla)
    if (s == "supressed all clazy checks") {} // NOLINT(clazy-*)
    if (s == "suppressed this check by name" && QDateTime::currentDateTime().toSecsSinceEpoch() < 0) {} // NOLINT(clazy-qstring-allocations)
}

void testClangTidySuppressionsNextLine()
{
    // NOLINTNEXTLINE
    QString s = "linting for line disabled";
    // NOLINTNEXTLINE(clazy-bla)
    if (s == "supressed nonexistent") {}
    // NOLINTNEXTLINE(clazy-*)
    if (s == "supressed all clazy checks") {}
    // NOLINTNEXTLINE(clazy-qstring-allocations)
    if (s == "suppressed this check by name" && QDateTime::currentDateTime().toSecsSinceEpoch() < 0) {}
}

void testRegionSupression()
{
    // NOLINTBEGIN
    QString s = "linting disabled";
    QString s1 = "linting disabled";
    QDateTime::currentDateTime().toSecsSinceEpoch();
    // NOLINTEND

    // NOLINTBEGIN(clazy-qdatetime-utc)
    QString warnme = "";
    const int suppressMe = QDateTime::currentDateTime().toSecsSinceEpoch();
    // NOLINTEND(clazy-qdatetime-utc)
}
