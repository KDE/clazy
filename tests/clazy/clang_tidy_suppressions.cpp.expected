clazy/clang_tidy_suppressions.cpp:8:9: warning: QString(const char*) being called [-Wclazy-qstring-allocations]
clazy/clang_tidy_suppressions.cpp:10:49: warning: Use QDateTime::currentSecsSinceEpoch() instead. It is significantly faster [-Wclazy-qdatetime-utc]
clazy/clang_tidy_suppressions.cpp:18:9: warning: QString(const char*) being called [-Wclazy-qstring-allocations]
clazy/clang_tidy_suppressions.cpp:22:49: warning: Use QDateTime::currentSecsSinceEpoch() instead. It is significantly faster [-Wclazy-qdatetime-utc]
clazy/clang_tidy_suppressions.cpp:34:22: warning: QString(const char*) being called [-Wclazy-qstring-allocations]
